presto-main/src/main/java/com/facebook/presto/operator/scalar/KleinbergBurstDetectionFunction.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.DoubleType;
+import com.facebook.presto.spi.type.IntegerType;
+import com.facebook.presto.spi.type.StandardTypes;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public final class KleinbergBurstDetectionFunction
+{
+    // Base of the exponential distribution that is used for modeling the
+    // event frequencies. Must be greater than 1.0, and determines the granularity
+    // of burst intensities detected (greater value, lower and more compact
+    // intensities).
+    private static final double DEFAULT_KLEINBERG_S = 2.0;
+    // Coefficient for the transition costs between states. Should be greater or
+    // equal than 1.0, although any value is accepted. High values make increases
+    // in the cost of intervals higher, amplifying big distances.
+    private static final double DEFAULT_KLEINBERG_GAMMA = 1.0;
+
+    @ScalarFunction
+    @Description("Return the Kleinberg burst detection for given timeseries")
+    @SqlType("array(int)")
+    @SqlNullable
+    public static Block kleinbergBurstDetection(@SqlType("array(double)") Block timeseries)
+    {
+        return kleinbergBurstDetection(timeseries, DEFAULT_KLEINBERG_S, DEFAULT_KLEINBERG_GAMMA);
+    }
+
+    @ScalarFunction
+    @Description("Return the Kleinberg burst detection for given timeseries, s and gamma")
+    @SqlType("array(int)")
+    @SqlNullable
+    public static Block kleinbergBurstDetection(
+            @SqlType("array(double)") Block timeseries,
+            @SqlType(StandardTypes.DOUBLE) double s,
+            @SqlType(StandardTypes.DOUBLE) double gamma)
+    {
+        if (timeseries.getPositionCount() <= 1) {
+            return null;
+        }
+        if (s <= 1.0) {
+            return null;
+        }
+        return new KleinbergBurstDetectionFunction().kleinberg(timeseries, s, gamma);
```

Comment:
Why `KleinbergBurstDetectionFunction:: kleinberg` isn't static?

Reply:
Made it static.